_posts/2018-02-05-chaining-http-requests-in-elm.md

```diff
@@ -0,0 +1,340 @@
+
+That's good but now we need to execute and chain these two requests, the second one depending on the result of the first one, where we retrieve the `actor.login` value of the event object.
+
+Elm is a pure language, meaning you can't have side effects in your functions (a side effect is when functions alter things outside of their scope and use these things: an HTTP request is a *huge* side effect). So your functions must return *something* that represents a given side effect, instead of executing it within the function scope itself. The Elm runtime will be in charge of actually performing the side effect.
+
+In Elm, you're usually gonna use a [Task] to describe future operations. Tasks may succeed or fail (like Promises do in JavaScript), but they need to be turned into [Elm commands] to be actually executed.
+
+So basically:
+
+1. We first craft `Http.Request`s,
+2. We turn them into `Task`s we can chain,
+3. We run these sequential tasks through a command (`Cmd msg`).
+
+The [Http] package provides `Http.toTask` to map an `Http.Request` into a `Task`. Let's use that here:
+
+```haskell
+fetchEvents : Task Http.Error (List String)
+fetchEvents =
+    eventsRequest |> Http.toTask
+
+fetchName : String -> Task Http.Error String
+fetchName login =
+    nameRequest login |> Http.toTask
+```
+
+I created these two simple functions mostly to focus on their return types; a `Task` must define an error type and a result type. For example, `fetchEvents` being an HTTP task, it will receive an `Http.Error` when the task fails, and a list of strings when the task succeeds.
+
+But dealing granularily with HTTP errors being out of scope of this blog post, and in order to keep things as less convoluted as possible, I'm gonna use `Task.mapError` to turn complex HTTP errors into their string representations:  
+
+```haskell
+toHttpTask : Http.Request a -> Task String a
+toHttpTask request =
+    request
+        |> Http.toTask
+        |> Task.mapError toString
+
+fetchEvents : Task String (List String)
+fetchEvents =
+    toHttpTask eventsRequest
+
+fetchName : String -> Task String String
+fetchName login =
+    toHttpTask (nameRequest login)
+```
+
+Here, `toHttpTask` is a helper turning an `Http.Request` into a `Task`, transforming the `Http.Error` complex type into a serialized, purely textual version of it: a `String`.
+
+We'll also need a function allowing to extract the very first element of a list, if any, as we did in JavaScript using `events[0]`. Such a function is builtin the `List` core module as `List.head`. And let's make this function a `Task` too, as that will ease chaining things alltogether and allow us to expose an error message when the list is empty:
```

Comment:
I know it doesn't make much sense, but it's `altogether` with a single `l`

Reply:
I also don't feel like `altogether` is the right word choice here. (It doesn't exactly mean "all together".) I think you want something like `chaining everything together`?